### ISSUE TITLE
Model is always changed

### ISSUE THREAD

**Issue by aaronchi** [devmynd/jsonb_accessor#30]
It appears that whenever a model is loaded, it registers a change event:

```
model = Model.last
model.changed?
=> true
```

**Comment by crismali** [comment]
@aaronchi good catch, definitely a bug.

**Comment by aaronchi** [comment]
I fixed it by adding `clear_changes_information` after the initialization method:

```
def _initialize_jsonb_attrs(jsonb_attribute, fields_map, jsonb_attribute_initialization_method_name)
        define_method(jsonb_attribute_initialization_method_name) do
          jsonb_attribute_hash = send(jsonb_attribute) || {}
          fields_map.names.each do |field|
            send("#{field}=", jsonb_attribute_hash[field.to_s])
          end
          clear_changes_information
        end
        after_initialize(jsonb_attribute_initialization_method_name)
      end
```

**Comment by crismali** [comment]
@aaronchi Playing around with this, `clear_changes_information` does ensure that the record doesn't falsely report that its changed, but it prevents changes from being persisted. We'll have to find another solution.

**Comment by aaronchi** [comment]
Seems to be working for me at the moment. Is there a reason you are using after_initialize here to set a value instead of just defining a getter method for each field?

**Comment by crismali** [comment]
What version of ActiveRecord are you using? I ran into the issue while
running the tests against version 4.0.

**Comment by QuotableWater7** [comment]
Still seeing an issue when I have a `jsonb_accessor` that wraps an array.  Has there been any activity on this?

**Comment by jhirn** [comment]
Hey everyone, we've actually been working on this issue internally. Sorry I forgot to provide an update. 

@aaronchi's suggestion does fix the issue with things tracking dirty but it uncovers a deeper problem where nested objects don't trigger the proper change set resulting in save silently having no effect. This would be a super subtle bug resulting in data loss. 

The underlying problem is buried deep within the `ActiveModel::Dirty` module and fixing the current code is something that's most likely going to require an overhaul of how we do nested models. We have an idea on how to fix this and hope to have something in place in the next couple of weeks. Apologies for the inconvenience and thanks for reporting the issues.

**Comment by QuotableWater7** [comment]
Thanks for update!

**Comment by jhirn** [comment]
Fixed by #42

**Comment by imton** [comment]
Hey, thanks for that. It solved some cases, still for **boolean** types, the bug stilss.
